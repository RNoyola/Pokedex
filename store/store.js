import { combineReducers, configureStore } from "@reduxjs/toolkit"
import { pokemonApi } from '../services/pokemon'
import catchReducer from "./slices/catchSlice"
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import AsyncStorage from "@react-native-async-storage/async-storage"

const persistConfig = {
  storage: AsyncStorage,
  key: "root",
  whitelist: ["catch"],
}

const rootReducer = combineReducers({
  [pokemonApi.reducerPath]: pokemonApi.reducer,
  catch: catchReducer
})


const persistedReducer = persistReducer(persistConfig, rootReducer)

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware(
      {
        serializableCheck: {
          ignoredActions: [ FLUSH ,REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        }
    }).concat(pokemonApi.middleware),
})

export const setupStore = preloadedState => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
    middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware(
      {
        serializableCheck: {
          ignoredActions: [ FLUSH ,REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        }
    }).concat(pokemonApi.middleware),
  })
}

export const persistor = persistStore(store)
