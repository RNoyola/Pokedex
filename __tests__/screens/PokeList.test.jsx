import React from 'react'
import { render, screen } from '@testing-library/react-native'
import Pokelist from '../../screens/PokeList'
import { Provider } from 'react-redux'
import { setupStore as store } from '../../store/store'
import { NavigationContainer } from '@react-navigation/native'
jest.useFakeTimers()

it('renders correctly', () => {
    const navigation = jest.fn()
    const wrapper = (
        <Provider store={store({})}>
          <NavigationContainer >
            <Pokelist navigation={navigation}/>
          </NavigationContainer>
        </Provider>
      )
    render(wrapper)
    expect(screen.toJSON()).toMatchSnapshot()
  });