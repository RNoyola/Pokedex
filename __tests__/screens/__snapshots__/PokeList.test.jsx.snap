// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#e11b2e",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "marginTop": 30,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "marginHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          {
            "flex": 1,
            "padding": 10,
          }
        }
        value=""
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 30,
            },
            {
              "justifyContent": "center",
              "padding": 5,
            },
            {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î¹¥
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderRadius": 6,
            "borderWidth": 1,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 20,
            "paddingHorizontal": 10,
            "width": "50%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "fontSize": 16,
                  "writingDirection": "ltr",
                },
                undefined,
                {},
              ]
            }
          >
            All
          </Text>
          <Image
            source={
              {
                "testUri": "../../react-native-element-dropdown/lib/commonjs/assets/down.png",
              }
            }
            style={
              {
                "height": 20,
                "tintColor": "gray",
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 6,
          "borderWidth": 3,
          "marginBottom": 150,
          "padding": 10,
        }
      }
    >
      <RCTScrollView
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
